### PR TITLE
[hue] Added configuration for port

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added configuration for port

Continues https://github.com/eclipse/smarthome/pull/6854

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by cweitkamp** [comment]
@martinvw Can you give me a hand on the `TlsTrustManager` implementations? For now I tries to follow the approach of the unifi binding including a "trust all" manager. Thanks.

The Philips Hue certificate looks like this - but I do not know if it is sufficient because the idea behind this PR is to allow the Hue binding to connect to similar Hue APIs like deCONZ too.
```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 6624557528005837 (0x1788fffe403ccd)
    Signature Algorithm: ecdsa-with-SHA256
        Issuer: C = NL, O = Philips Hue, CN = 001788fffe403ccd
        Validity
            Not Before: Jan  1 00:00:00 2017 GMT
            Not After : Jan  1 00:00:00 2038 GMT
        Subject: C = NL, O = Philips Hue, CN = 001788fffe403ccd
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:64:1f:2f:c1:10:62:bf:fc:f1:aa:9d:41:57:e8:
                    46:db:d1:49:9d:f0:6c:02:cf:cf:70:9e:ec:2f:9e:
                    d4:e3:e7:ce:81:c4:20:a3:f2:2a:af:75:3a:be:c7:
                    7b:58:f4:4a:f8:71:e8:3b:24:a1:2a:19:52:1d:91:
                    4b:01:41:14:e2
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier: 
                38:C8:28:A2:62:A4:AC:2A:53:83:21:4D:EF:67:2D:D3:68:EE:C5:AB
            X509v3 Authority Key Identifier: 
                keyid:38:C8:28:A2:62:A4:AC:2A:53:83:21:4D:EF:67:2D:D3:68:EE:C5:AB
                DirName:/C=NL/O=Philips Hue/CN=001788fffe403ccd
                serial:17:88:FF:FE:40:3C:CD

            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
    Signature Algorithm: ecdsa-with-SHA256
         30:46:02:21:00:e4:f8:51:c4:84:80:a6:c6:4a:95:38:b2:f2:
         e3:e4:79:56:b4:b0:32:6e:27:e3:56:61:84:c5:4b:f4:85:48:
         3a:02:21:00:ec:f3:07:d5:f7:53:8a:90:83:11:ba:5e:e0:e5:
         d3:89:de:4e:8e:d4:95:7c:0b:ad:e1:a6:48:17:06:d4:8d:40
```

File: addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java [hunk 1]
```diff
@@ -534,6 +535,7 @@ private void updateBridgeThingConfiguration(String userName) {
     }
 
     private void handleAuthenticationFailure(Exception ex, String userName) {
+        logger.debug("{}", ex);
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java]: To be removed after we found a solution for the occurring exception is found.

**Comment by davidgraeff** [comment]
Before we run into merge conflicts at some point, can we merge this? At least the port fix looks good and https needs further tests, I agree. But better having that in the main tree than rotting, I guess?

**Comment by cweitkamp** [comment]
I thought about splitting this up into two separate PRs. One for the port and one for the HTTPS connection. I'll be the nicest way.

**Comment by cweitkamp** [comment]
@davidgraeff I removed all HTTPS related changes which were visible for the user or obsolete code.

File: addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridge.java [hunk 2]
```diff
@@ -998,24 +1005,25 @@ public void handleErrors(Result result) throws IOException, ApiException {
...
         if (username != null) {
-            relativeUrl += "/" + enc(username);
+            basePath += "/" + enc(username);
         }
-        if (!path.isEmpty()) {
-            relativeUrl += "/" + path;
+        try {
+            URI uri = new URI(protocol, null, ip, port, path.isEmpty() ? basePath : basePath + "/" + path, null, null);
+            return uri.toString();
+        } catch (URISyntaxException e) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridge.java]: Should we do some logging here? Its unlikely that the url is correct?

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridge.java]: Yes, I will. And I will do a second change: I will move the creation of the `URI` into the constructor. It is sufficient to resolve it once and not every time we want to connect to the Hue API.

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/HueBridge.java]: Done. But let me test it for at least one day before we merge.

**Comment by cweitkamp** [comment]
> But let me test it for at least one day before we merge.

It looks fine. Works smoothly without unexpected errors. I will give it a go.

**Comment by davidgraeff** [comment]
I'll merge it in then. The hue binding is really causing havoc on my system because of it constantly forgetting the port at the moment (and colliding with hue emulation therefore).

**Review by Germania1989 — COMMENTED** [review]
N

**Pull request merged**